src/styles/transitions/BaseTransition.js

```diff
@@ -0,0 +1,39 @@
+import { CSSTransition } from 'react-transition-group';
+import PropTypes from 'prop-types';
+import React from 'react';
+import useDelayedChange from 'lib/effects/useDelayedChange';
+
+export const BaseTransitionPropTypes = {
+  in: PropTypes.bool,
+  children: PropTypes.node.isRequired,
+  timeout: PropTypes.oneOfType([
+    PropTypes.number,
+    PropTypes.shape({
+      appear: PropTypes.number,
+      enter: PropTypes.number,
+      exit: PropTypes.number,
+    }),
+  ]),
+};
+
+const BaseTransition = ({ children, in: inProp, ...props }) => {
+  const delayedInProp = useDelayedChange(inProp, 500);
+
+  return (
+    <CSSTransition in={delayedInProp} mountOnEnter unmountOnExit {...props}>
+      {children}
+    </CSSTransition>
+  );
+};
+
+BaseTransition.defaultProps = {
+  in: false,
+  timeout: 200,
+};
```

Comment:
hey @axbarsan  ¿Is there any reason why you wrote this instead of using `transition-delay?

I am reading this and don't understand the purpose of this...

Reply:
Since in most places we do our animations something like "set this state variable to `true` and then set it back to `false`". Because that is so quick, React merges the state changes and only does the state change to `false`, so the `enter` animation never happens, just the `exit` one.

What I've done is I've pretty much load balanced the animation `in` property change. This is similar to the concept of debouncing, but this one just delays the `in` property changes, where debouncing merges them.
After each change of the `in` property, it waits `500ms`, and if there is another change happening, it will postpone it to when `500ms` have passed from the last change. If the `500ms` have passed without any change, the next change will execute immediately, without any delay.

Reply:
This pretty much makes us not need to use `setTimeout` in every component we want to do these types of animations

Reply:
does that effectively limit our time between "in and out" animations to being 500ms 

So like a checkmark can appear quickly, hold for a while, and then disappear quickly.

But that 'hold for a while', cant be any shorter than 500ms right?

I'm generally ok with that. Just getting an understanding.